backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/radiobutton/BpkRadioButton.kt

```diff
@@ -22,59 +22,60 @@ import androidx.compose.foundation.clickable
 import androidx.compose.foundation.layout.Arrangement
 import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.RowScope
 import androidx.compose.material.ExperimentalMaterialApi
+import androidx.compose.material.LocalContentAlpha
+import androidx.compose.material.LocalContentColor
 import androidx.compose.material.LocalMinimumTouchTargetEnforcement
+import androidx.compose.material.LocalTextStyle
 import androidx.compose.material.RadioButton
 import androidx.compose.material.RadioButtonDefaults
 import androidx.compose.runtime.Composable
 import androidx.compose.runtime.CompositionLocalProvider
 import androidx.compose.runtime.remember
 import androidx.compose.ui.Alignment
+import androidx.compose.ui.ExperimentalComposeUiApi
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.semantics.Role
+import androidx.compose.ui.semantics.invisibleToUser
+import androidx.compose.ui.semantics.semantics
 import net.skyscanner.backpack.compose.text.BpkText
 import net.skyscanner.backpack.compose.theme.BpkTheme
 import net.skyscanner.backpack.compose.tokens.BpkColor
 import net.skyscanner.backpack.compose.tokens.BpkDimension
 import net.skyscanner.backpack.compose.utils.dynamicColorOf
 
-@OptIn(ExperimentalMaterialApi::class)
 @Composable
 fun BpkRadioButton(
   selected: Boolean,
+  text: String,
   onClick: (() -> Unit)?,
   modifier: Modifier = Modifier,
   enabled: Boolean = true,
   interactionSource: MutableInteractionSource = remember { MutableInteractionSource() },
 ) {
-  // our design system isn't designed with the minimum touch target in mind at the moment.
-  // Disable the enforcement to avoid the extra padding
-  CompositionLocalProvider(LocalMinimumTouchTargetEnforcement provides false) {
-    RadioButton(
-      selected = selected,
-      onClick = onClick,
-      modifier = modifier,
-      enabled = enabled,
-      interactionSource = interactionSource,
-      colors = RadioButtonDefaults.colors(
-        selectedColor = BpkTheme.colors.primary,
-        unselectedColor = BpkTheme.colors.textSecondary,
-        disabledColor = BpkColor.SkyGrayTint04,
-      ),
-    )
+  BpkRadioButton(
+    selected = selected,
+    onClick = onClick,
+    modifier = modifier,
+    enabled = enabled,
+    interactionSource = interactionSource
+  ) {
+    BpkText(text = text)
   }
 }
 
 @Composable
 fun BpkRadioButton(
   selected: Boolean,
-  text: String,
   onClick: (() -> Unit)?,
   modifier: Modifier = Modifier,
   enabled: Boolean = true,
   interactionSource: MutableInteractionSource = remember { MutableInteractionSource() },
+  content: @Composable RowScope.(Boolean) -> Unit,
 ) {
   val rowModifier = if (onClick != null) {
-    modifier.clickable(interactionSource = interactionSource, indication = null, onClick = onClick)
+    modifier.clickable(interactionSource = interactionSource, indication = null, role = Role.RadioButton, onClick = onClick)
```

Comment:
this makes the row the radiobutton as well - the announcement will include the text in the content, as well as the toggle behaviour/state

Reply:
Does it work okay with VoiceOver?

Reply:
yep - as i said it'll announce the content with the role & call to action to toggle (and announces the state when toggle is changed)